pool/recycle.go

```diff
@@ -0,0 +1,52 @@
+package pool
```

Comment:
Nice, now I have a spot to move the existing `lnwallet.SigPool` struct to! Probably something like `pool.Sig`.